common/src/backends/otlp_tracing_bridge.rs

```diff
@@ -152,3 +153,104 @@ const fn severity_of_level(level: &Level) -> Severity {
         Level::ERROR => Severity::Error,
     }
 }
+
+pub struct ErrorTracingLayer<S> {
+    _registry: std::marker::PhantomData<S>,
+}
+
+impl<S> ErrorTracingLayer<S>
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+{
+    pub fn new() -> Self {
+        ErrorTracingLayer {
+            _registry: std::marker::PhantomData,
+        }
+    }
+}
+
+impl<S> Layer<S> for ErrorTracingLayer<S>
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+{
+    fn on_event(
+        &self,
+        event: &tracing_core::Event<'_>,
+        ctx: tracing_subscriber::layer::Context<'_, S>,
+    ) {
+        // We only care about error events.
+        if !ErrorVisitor::is_valid(event.metadata()) {
+            return;
+        }
+
+        let mut visitor = ErrorVisitor::default();
+        event.record(&mut visitor);
+
+        let DatadogError {
+            message,
+            r#type,
+            stack,
+        } = visitor.error;
+
+        if let Some(span) = ctx.lookup_current() {
+            if let Some(otel_data) = span.extensions_mut().get_mut::<OtelData>() {
+                let error_fields = [
+                    KeyValue::new("error.message", message),
+                    KeyValue::new("error.type", r#type),
+                    KeyValue::new("error.stack", stack),
+                ];
+                let builder_attrs = otel_data
+                    .builder
+                    .attributes
+                    .get_or_insert(Vec::with_capacity(3));
+                builder_attrs.extend(error_fields);
```

Comment:
nit: out of curiosity, after getting the `span` can't we set the same fields by using the `.set_attribute` function? Just wondering if it can be done simpler.

Reply:
The trait I used in the utility function is [implemented for `tracing::Span`](https://docs.rs/tracing-opentelemetry/latest/tracing_opentelemetry/trait.OpenTelemetrySpanExt.html#impl-OpenTelemetrySpanExt-for-Span), but the type here is [SpanRef.](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/registry/struct.SpanRef.html)

Reply:
I see, thanks!